### ISSUE TITLE
Binary segmentation on a custom dataset 

### ISSUE THREAD

**Issue by panovr** [yassouali/pytorch-segmentation#139]
Hi, my dataset has images and corresponding masks. The dataset folders just like below:
```
Dataset
├── images
│   ├── train├── 0.png
│            ├── 1.png
│            └......
│   ├── val├── 0.png
│          ├── 1.png
│          └......
├── labels
│   ├── train├── 0.png
│            ├── 1.png
│   ├── val├── 0.png
│          ├── 1.png
│          └......
```
The dataset has only two class: background and foreground, so it is a binary segmentation. The mask image is a 3-channel image like below.  This mask is a black/white 3-channel image, and the black color (r/g/b=0/0/0) indicates background, and the white color (r/g/b=255/255/255) indicates foreground. 
![12](https://user-images.githubusercontent.com/319796/187066887-eef726cd-6893-4837-87ec-71901de96b1c.png)
My dataset code is:
```
class MyDataset(BaseDataSet):
    def __init__(self, **kwargs):
        self.num_classes = 2
        super(MyDataset, self).__init__(**kwargs)

    def _set_files(self):
        if self.split in  ["train", "val"]:
            self.image_dir = os.path.join(self.root, 'images', self.split)
            self.label_dir = os.path.join(self.root, 'labels', self.split)
            self.files = [os.path.basename(path).split('.')[0] for path in glob(self.image_dir + '/*.png')]
        else: raise ValueError(f"Invalid split name {self.split}")
    
    def _load_data(self, index):
        image_id = self.files[index]
        image_path = os.path.join(self.image_dir, image_id + '.png')
        label_path = os.path.join(self.label_dir, image_id + '.png')
        image = np.asarray(Image.open(image_path).convert('RGB'), dtype=np.float32)
        label = cv2.imread(label_path, cv2.IMREAD_GRAYSCALE)
        return image, label, image_id

class MD(BaseDataLoader):
    def __init__(self, data_dir, batch_size, split, crop_size=None, base_size=None, scale=True, num_workers=1, val=False,
                    shuffle=False, flip=False, rotate=False, blur= False, augment=False, val_split= None, return_id=False):

        self.MEAN = [0.48897059, 0.46548275, 0.4294]
        self.STD = [0.22861765, 0.22948039, 0.24054667]

        kwargs = {
            'root': data_dir,
            'split': split,
            'mean': self.MEAN,
            'std': self.STD,
            'augment': augment,
            'crop_size': crop_size,
            'base_size': base_size,
            'scale': scale,
            'flip': flip,
            'blur': blur,
            'rotate': rotate,
            'return_id': return_id,
            'val': val
        }

        self.dataset = MyDataset(**kwargs)
        super(MD, self).__init__(self.dataset, batch_size, shuffle, num_workers, val_split)
```
My config file is:
```
"name": "PSPNet",
    "n_gpu": 1,
    "use_synch_bn": false,

    "arch": {
        "type": "PSPNet",
        "args": {
            "backbone": "resnet50",
            "freeze_bn": false,
            "freeze_backbone": false
        }
    },

    "train_loader": {
        "type": "MD",
        "args":{
            "data_dir": "data/md",
            "batch_size": 8,
            "base_size": 512,
            "crop_size": 380,
            "augment": true,
            "shuffle": true,
            "scale": true,
            "flip": true,
            "rotate": true,
            "blur": false,
            "split": "train",
            "num_workers": 2
        }
    },

    "val_loader": {
        "type": "md",
        "args":{
            "data_dir": "data/md",
            "batch_size": 8,
            "crop_size": 480,
            "val": true,
            "split": "val",
            "num_workers": 2
        }
    },

    "optimizer": {
        "type": "SGD",
        "differential_lr": true,
        "args":{
            "lr": 0.01,
            "weight_decay": 1e-4,
            "momentum": 0.9
        }
    },

    "loss": "CrossEntropyLoss2d",
    "ignore_index": -1,
    "lr_scheduler": {
        "type": "Poly",
        "args": {}
    },

    "trainer": {
        "epochs": 10,
        "save_dir": "saved/",
        "save_period": 1,
  
        "monitor": "max Mean_IoU",
        "early_stop": 2,
        
        "tensorboard": true,
        "log_dir": "saved/runs",
        "log_per_iter": 2,

        "val": true,
        "val_per_epochs": 2
    }
}
```
For this binary segmentation problem, my question is:
* How to set the ` ignore_index` for my dataset? Because my mask is a black/white 3-channel image, and the black color indicates background, and the white color indicates foreground. (I understand that `ignore_index=255` for pascal voc because pascal voc uses `void` label (index 255) to indicate border region)
* How to set `num_classes`, should it be 1 or 2?
* How to set `palette` for my dataset?
Thanks!

**Comment by yassouali** [comment]
@panovr Hi, sorry for the delayed response;

I think the first thing is to convert the masks from 3D to 2D (a simple 2d array with 0/1 indicating each class), and if in your case you don't need to use ignore_class, then you can just keep ignore_index=255 since it won't have any effect.

palette is just used to map a label back to color to show it, you can set your to [0,0,0, 255, 255, 255] so that you show back for class 0 and white for class 1

Hope this helps;
